extensions/amp-form/0.1/test/test-amp-form.js

```diff
@@ -2354,53 +2383,114 @@ describes.repeated(
         });
 
         describe('Var Substitution', () => {
-          it('should substitute hidden fields variables in XHR async', () => {
-            return getAmpForm(getForm()).then((ampForm) => {
-              const form = ampForm.form_;
-              const clientIdField = createElement('input');
+          describe('basic XHR request', () => {
```

Comment:
There are more featuers to test. A way to ensure that we run each tests in two forms would be ideal:

(super abstract, doesn't need this signature)

```js
const inside = html`
  <form>
    <input />
  </form>
`;
const outside = html`
  <div>
    <input form="foo" />
    <form id="foo">...</form>
  </div>
`;
myTestCase1(inside, inside.querySelector('input'));
myTestCase2(outside, inside.querySelector('input'));
```

I also think we should try to keep the structure of existing tests as much as possible, and reuse them in a way that makes it clear that we're testing the same behavior in two diff tree modes.